classes/Domain/Speaker/SpeakerProfile.php

```diff
@@ -63,9 +64,9 @@ public function needsProfile(): bool
      *
      * @throws NotAllowedException
      *
-     * @return Talk[]
+     * @return Collection|Talk[]
      */
-    public function getTalks(): array
+    public function getTalks(): Collection
```

Comment:
How about we leave it to `array` and instead enforce that an array is returned?

This has the advantage that we don't expose the collection, and thus, it can't be modified.